objects/adffc0/contained/706d9f/contained/4363a8/script.lua

```diff
@@ -0,0 +1,299 @@
+    B1 = {x = -4.79, z = 5.20},
+    B2 = {x = -8.13, z = -1.50},
+    B3 = {x = -9.75, z = -7.24},
+    B4 = {x = -0.79, z = -4.11},
+    B5 = {x = 1.81, z = -0.19},
+    B6 = {x = 2.49, z = 5.46},
+    B7 = {x = 6.07, z = -3.67},
+    B8 = {x = 10.34, z = 6.36},
+    C1 = {x = -4.53, z = 5.06},
+    C2 = {x = -7.18, z = -1.77},
+    C3 = {x = -10.04, z = -6.53},
+    C4 = {x = 1.80, z = -6.31},
+    C5 = {x = 1.33, z = -1.13},
+    C6 = {x = 3.29, z = 5.16},
+    C7 = {x = 7.43, z = -0.98},
+    C8 = {x = 9.96, z = 5.25},
+    D1 = {x = 2.71, z = 7.17},
+    D2 = {x = -6.25, z = -0.88},
+    D3 = {x = -10.28, z = -5.88},
+    D4 = {x = -3.21, z = -5.49},
+    D5 = {x = 0.37, z = 1.01},
+    D6 = {x = 3.81, z = -3.79},
+    D7 = {x = 6.22, z = 0.54},
+    D8 = {x = 10.36, z = 5.77},
+    E1 = {x = -2.66, z = 6.38},
+    E2 = {x = -7.08, z = -0.15},
+    E3 = {x = -10.15, z = -6.00},
+    E4 = {x = -0.32, z = -5.32},
+    E5 = {x = -1.72, z = -0.71},
+    E6 = {x = 6.25, z = -2.74},
+    E7 = {x = 5.51, z = 3.56},
+    E8 = {x = 9.99, z = 3.45},
+    F1 = {x = -3.35, z = 6.13},
+    F2 = {x = -6.58, z = -1.59},
+    F3 = {x = -9.20, z = -5.25},
+    F4 = {x = -0.53, z = -5.15},
+    F5 = {x = 1.65, z = 2.34},
+    F6 = {x = 4.90, z = 5.66},
+    F7 = {x = 6.16, z = -3.74},
+    F8 = {x = 7.19, z = 1.27},
+    G1 = {x = -2.98, z = 5.50},
+    G2 = {x = -5.80, z = -1.40},
+    G3 = {x = -8.70, z = -6.87},
+    G4 = {x = -1.70, z = -4.97},
+    G5 = {x = 1.18, z = -0.43},
+    G6 = {x = 3.59, z = 5.61},
+    G7 = {x = 5.63, z = -3.58},
+    G8 = {x = 9.13, z = 4.32},
+    H1 = {x = -2.64, z = 6.01},
+    H2 = {x = -6.60, z = 0.10},
+    H3 = {x = -10.38, z = -6.63},
+    H4 = {x = 1.11, z = -6.46},
+    H5 = {x = -0.02, z = -2.73},
+    H6 = {x = 1.87, z = 2.69},
+    H7 = {x = 7.68, z = 1.08},
+    H8 = {x = 8.45, z = 6.87},
+}
+
+local thematicPositions = {
+}
+
+function err(color, message)
+    Player[color].broadcast(message, Color.Red)
+end
+
+function castDownCallback()
+    -- If we're not currently casting down, return immediately to avoid overhead.
+    if not Global.getVar("castDown") then
+        return
+    end
+    
+    local board = getObjectFromGUID(Global.getVar("castDown"))
+    local drowningTiles = getBoardTiles(board)
+    for _,guid in pairs(drowningTiles) do
+        local obj = getObjectFromGUID(guid)
+        if obj ~= nil then
+            obj.destruct()
+        end
+    end
+end
+
+-- Starts the Cast Down timer.
+function startCastDownTimer()
+    -- We store the reference to the Cast Down timer in Global, so every Deeps token has access to it.
+    -- Every Deeps token will try to do this, so if it already exists, don't worry about it.
+    if Global.getVar(timerVarName) then
+        return
+    end
+    -- Cast Down lasts 0.5 seconds.
+    -- We need to drop the tiles early, to give things time to fall, so do this every 0.2 seconds.
+    Wait.time(castDownCallback, 0.2, -1)
+    Global.setVar(timerVarName, true)
```

Comment:
what is in charge of setting this back to false once castdown operation is done?

Reply:
This timer doesn't only run during a cast down: it's running *permanently*. Its job is to detect when a cast down is in progress and delete the tiles associated with that board. It needs to run so frequently (five times per second), so that it can catch any cast down operation early, allowing the tiles to be deleted quickly, allowing anything on top of them time to drop and hit the cast down board while the cast down operation is still in progress.

The point of the boolean is that this timer is only created once, rather than being created afresh by every single Deeps token.

Thankfully, this mess is simply a holdover from being part of an additive mod, and thus all this code having to exist on the Deeps tokens. The entire timer can be removed just by making deleting the drowning tiles the job of the cast down code itself, so I'll do that.